axelrod/mock_player.py

```diff
@@ -1,53 +1,74 @@
-import copy
-from axelrod import get_state_distribution_from_history, Player, \
-    update_history, update_state_distribution,  Actions
+import warnings
+from axelrod import Actions, Player, update_history, update_state_distribution
 
 C, D = Actions.C, Actions.D
 
 
 class MockPlayer(Player):
-    """Creates a mock player that enforces a particular next move for a given
-    player."""
+    """Creates a mock player that copies a history and state distribution to
+    simulate a history of play, and then plays a given sequence of actions. If
+    no actions are given, plays like Cooperator.
+    """
+
+    name = "Mock Player"
 
-    def __init__(self, player, move):
+    def __init__(self, actions=None, history=None, state_dist=None):
         # Need to retain history for opponents that examine opponents history
         # Do a deep copy just to be safe
         super().__init__()
-        self.history = copy.deepcopy(player.history)
```

Comment:
The new code below it copies the list element by element using `update_history`.